test/net/sourceforge/kolmafia/session/LimitModeTest.java

```diff
@@ -394,4 +433,175 @@ void someLimitMCD(final LimitMode lm) {
       assertThat(lm.limitMCD(), is(true));
     }
   }
+
+  @Nested
+  class Recovery {
+    @ParameterizedTest
+    @EnumSource(
+        value = LimitMode.class,
+        names = {"NONE", "ASTRAL", "BIRD", "MOLE", "ROACH"})
+    void someDontLimitRecovery(final LimitMode lm) {
+      assertThat(lm.limitRecovery(), is(false));
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+        value = LimitMode.class,
+        names = {"SPELUNKY", "BATMAN", "ED"})
+    void someLimitRecovery(final LimitMode lm) {
+      assertThat(lm.limitRecovery(), is(true));
+    }
+  }
+
+  // *** Pseudo LimitModes
+
+  @Nested
+  class Astral {
+    @AfterEach
+    public void afterEach() {
+      KoLConstants.activeEffects.clear();
```

Comment:
this should be unnecessary due to the cleanups that add and clear effects?

Reply:
Ditto.